src/components/common/Swap/components/CurrencyBox/CurrencyBox.tsx

```diff
@@ -45,19 +47,19 @@ const CurrencyBox = ({ value, coin, mode, balance, setAmount, loading, onCoinSel
   const handleMaxClick = useCallback(() => {
     let amountStringToSet;
     if (coin === "ETH" && mode === "sell") {
-      const amountWithoutGasFee = balance - MinEthValue;
-      amountStringToSet = amountWithoutGasFee > 0 ? amountWithoutGasFee.toFixed(9) : balance.toString();
+      const amountWithoutGasFee = balance.sub(MinEthValueBN);
+      amountStringToSet = amountWithoutGasFee.gt(0)
+        ? amountWithoutGasFee.formatUnits(decimals)
```

Comment:
Is it the same as `format({ units:`? Should we keep just one style?

Reply:
Fixed